From 1557b7bd45499c738d17a97dbb1713a59faddc1f Mon Sep 17 00:00:00 2001
From: Eric Garver <eric@garver.life>
Date: Mon, 1 Jul 2024 15:07:27 -0400
Subject: [PATCH 25/31] v2.3.0: test(functions): fix iptables normalization for
 opt field

The sed expression didn't consume multiple spaces. All of the
normalization nuances are stacking. :(

@@ -3,5 +3,5 @@
 IN_public_deny 0 -- ::/0 ::/0
 IN_public_allow 0 -- ::/0 ::/0
 IN_public_post 0 -- ::/0 ::/0
-ACCEPT 58 -- ::/0 ::/0
+ACCEPT 58 -- -- ::/0 ::/0

Fixes: 9a11ab13c109 ("test(functions): ip6tables: normalize opt field output")
(cherry picked from commit e3f44a77be2f64be1beb7441314c37a9d773c063)
---
 src/tests/functions.at | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tests/functions.at b/src/tests/functions.at
index d1c89ed5b982..8b07908c667c 100644
--- a/src/tests/functions.at
+++ b/src/tests/functions.at
@@ -423,7 +423,7 @@ m4_define([IPXTABLES_LIST_RULES_NORMALIZE], [dnl
     dnl commit 6e41c2d8747b ("iptables: xshared: Output '--' in the opt field
     dnl in ipv6's fake mode") added "--" to the output of ip6tables. This regex
     dnl adds the "--" for older ip6tables versions.
-        -e ['s/^\([-a-zA-Z0-9_]\+\)[ ]\([^ ]*\)[ ][-]\?[-]\?[ ]\?/\1 \2 -- /g'] dnl
+        -e ['s/^\([-a-zA-Z0-9_]\+\)[ ]\+\([^ ]*\)[ ]\+[-]\?[-]\?[ ]\?/\1 \2 -- /g'] dnl
 ])
 
 m4_define([IPTABLES_LIST_RULES_NORMALIZE], [dnl
-- 
2.48.1

